### PR TITLE
Strict flags and GIR docs enabled by default.

### PR DESCRIPTION
**Pull request by BlackVegetable**
The configure.ac file now instructs Autoconf to set the default values to true
for the flags --enable-strict-flags and --enable-gir-doc.

[endlessm/eos-sdk#1978]

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -48,4 +48,11 @@ gtkdocize || exit $?
 autoreconf -fi || exit $?
 
 cd "$olddir"
-test -n "$NOCONFIGURE" || "$srcdir/configure" "$@"
+
+# If no arguments passed, will use default arguments for eos-metrics.
+if [ $# -eq 0 ]
+    then
+        test -n "$NOCONFIGURE" || "$srcdir/configure" "--enable-strict-flags" "--enable-gtk-doc" "--enable-gir-doc"
```
Comment (ptomato) [inline autogen.sh]: I would prefer to change this in `configure.ac` by making "yes" the default for each of these options instead of "no". People won't think to look in this file because it's not customary to add arguments here; let's keep them in one place.

Comment (Kurt-von-Laven) [inline autogen.sh]: This doesn't handle the case where someone runs configure directly. Recommend instead removing the flags entirely and modify configure.ac to always configure the way that it currently does when passed these flags. @ptomato is there any reason we would ever want to compile without strict flags, GTK, or GIR enabled? If so, we could change the flags to be "--disable-*" and flip the default behavior in configure.ac.

Comment (BlackVegetable) [inline autogen.sh]: Bahahaha, simultaneous comments.  Thanks, I agree with your logic(s), and will make that change.

Comment (ptomato) [inline autogen.sh]: @Kurt-von-Laven Autoconf already takes care of the `--disable-` versions of the arguments. There definitely is a use case for turning off `gtk-doc` and `gir-doc`, which is that they slow down the build considerably. I'm not sure about `strict-flags` other than that it's generally bad practice to force particular flags on a developer without a way to override it.

Comment (Kurt-von-Laven) [inline autogen.sh]: Makes sense. Thanks Philip!

File: configure.ac [hunk 2]
```diff
@@ -182,7 +182,7 @@ AC_SUBST(STRICT_CFLAGS)
 # during 'make distcheck'.
 AC_ARG_ENABLE([gir-doc],
     [AS_HELP_STRING([--enable-gir-doc],
-        [Build GIR documentation for Javascript @<:@default=no@:>@])])
+        [Build GIR documentation for Javascript @<:@default=yes@:>@])])
```
Comment (Kurt-von-Laven) [inline configure.ac]: Also need to pass the last two parameters to AC_ARG_ENABLE the same way they are passed above for strict-flags (c.f., https://autotools.io/autoconf/arguments.html).

File: configure.ac [hunk 3]
```diff
@@ -146,9 +146,9 @@ AC_CACHE_SAVE
 # during 'make distcheck'.
```
Comment (Kurt-von-Laven) [inline configure.ac]: Also need to remove https://github.com/endlessm/eos-metrics/blob/master/Makefile.am#L47 and the following line.

Comment (Kurt-von-Laven) [inline configure.ac]: Also need to update https://github.com/endlessm/eos-metrics/blob/master/Makefile.am#L64.

Comment (Kurt-von-Laven) [inline configure.ac]: In general, I recommend using git grep to catch dangling references like this.

Comment (BlackVegetable) [inline configure.ac]: Good point.  I'm pretty new to automake so I feel pretty lobotomized when working on it.

File: Makefile.am [hunk 4]
```diff
@@ -43,9 +43,7 @@ AM_CFLAGS = @STRICT_CFLAGS@
 # Make sure to run Gtk-doc tests and build the documentation when doing
 # 'make distcheck'
 AM_DISTCHECK_CONFIGURE_FLAGS = \
-	--enable-gtk-doc \
-	--enable-gir-doc \
-	--enable-strict-flags \
+	--enable-gtk-doc
```
Comment (Kurt-von-Laven) [inline Makefile.am]: Still need the `\` at the end of the line since we still have `$(NULL)` below. If you wanted to put everything on one line, you could do: `AM_DISTCHECK_CONFIGURE_FLAGS = $(NULL)`.

File: Makefile.am [hunk 5]
```diff
@@ -43,9 +43,7 @@ AM_CFLAGS = @STRICT_CFLAGS@
 # Make sure to run Gtk-doc tests and build the documentation when doing
 # 'make distcheck'
 AM_DISTCHECK_CONFIGURE_FLAGS = \
```
Comment (Kurt-von-Laven) [inline Makefile.am]: Still need to update https://github.com/endlessm/eos-metrics/blob/master/Makefile.am#L64.

File: configure.ac [hunk 6]
```diff
@@ -182,7 +182,9 @@ AC_SUBST(STRICT_CFLAGS)
...
 AC_ARG_ENABLE([gir-doc],
     [AS_HELP_STRING([--enable-gir-doc],
-        [Build GIR documentation for Javascript @<:@default=no@:>@])])
+        [Build GIR documentation for Javascript @<:@default=yes@:>@])],
```
Comment (Kurt-von-Laven) [inline configure.ac]: Optional (pre-existing defect): "Javascript" --> "JavaScript"

File: Makefile.am (hunk 4 above)
Comment (ptomato) [inline Makefile.am]: Don't do `VARIABLE = $(NULL)`, do `VARIABLE =` instead. The `$(NULL)` is only there at the end of multiline lists so that you don't accidentally put a backslash after your last item and combine it with whatever unrelated code comes after your list.

File: Makefile.am [hunk 7]
```diff
@@ -43,9 +43,7 @@ AM_CFLAGS = @STRICT_CFLAGS@
 # Make sure to run Gtk-doc tests and build the documentation when doing
```
Comment (Kurt-von-Laven) [inline Makefile.am]: I think the commit message and pull request description are stale. When I run `./autogen.sh --enable-strict-flags --enable-gtk-doc --prefix=/usr && make && make check && make distcheck` on master at HEAD in eos-metrics, I hit the following error during the `make distcheck`. Also, docs/reference/eosmetrics-js is only made when I explicitly configure with `--enable-gir-doc`. That is why I don't think that GIR doc is enabled by default (without your changes in this pull request).

```
make[1]: se ingresa al directorio `/home/endless/endlessm/eos-metrics'
***
*** You must configure with --enable-gtk-doc and --enable-gir-doc to
*** run make dist or make distcheck.
***
make[1]: *** [dist-hook] Error 1
make[1]: se sale del directorio `/home/endless/endlessm/eos-metrics'
make: *** [distdir] Error 2
```

File: Makefile.am (hunk 4 above)
Comment (Kurt-von-Laven) [inline Makefile.am]: Sorry, that was a typo. I meant to say that you could do `AM_DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc` if you wanted to put everything on one line. We don't want `AM_DISTCHECK_CONFIGURE_FLAGS` to be empty.

**Comment by Kurt-von-Laven** [comment]
**CR Pass** Thanks for finishing this one off, Patrick.
